### ISSUE TITLE
Test source does not appear to work

### ISSUE THREAD

**Issue by srcejon** [f4exb/sdrangel#661]
Hi,

It looks like the Test Source does not appear to work. If I press start acquisition, there's no spectrum/waterfall (or anything in channel analyser scope) regardless of test source settings. Have tried both on Windows and Linux with SDRangel 4.20.

**Comment by chris4git** [comment]
Hi,

same with SDRangel 5.14.0/Ubuntu 20.04

**Comment by f4exb** [comment]
I don't know for how long it has been like this but this seems to be due to the new worker handling:
```
2020-11-01 10:06:34.934 (D) TestSourceWorker::startWork
2020-11-01 10:06:34.934 (W) QObject::startTimer: Timers cannot be started from another thread
```
The solution is to move timer start and stop in constructor and destructor respectively. Stream generation is conditional to `m_running` anyway.

**Comment by f4exb** [comment]
Implemented in v4.21.1 and 5.15.1
